test/parallel/test-writeuint.js

```diff
@@ -122,6 +122,19 @@ function test32(clazz) {
 }
 
 
+function testUint(clazz) {
+  const data = new clazz(8);
+
+  ASSERT.throws(function() {
+    data.writeUIntBE(0x100, 0, 1);
+  }, /value is out of bounds/);
+  ASSERT.throws(function() {
+    data.writeUIntLE(0x100, 0, 1);
+  }, /value is out of bounds/);
```

Comment:
Maybe add checks for byteLength > 1?  If someone changed the logic from `Math.pow(2, 8 * byteLength)` to `1 << (byteLength << 3)` (which looks innocuous at first glance), it would start doing the wrong thing for byteLength == 4.